### PR TITLE
Add request metadata to return signature #22

### PR DESCRIPTION
**Pull request by jaredkerim**

### CODE REVIEW

File: embedly/embedly/extract.py [hunk 1]
```diff
@@ -23,7 +27,12 @@ def _get_cached_url(self, url):
         cached_data = self.redis_client.get(cache_key)
 
         if cached_data is not None:
-            return json.loads(cached_data)
+            try:
+                return json.loads(cached_data)
+            except ValueError:
+                raise URLExtractorException(
```
Comment (jaredkerim) [inline embedly/embedly/extract.py]: If we're unable to read the data for a URL from the cache, we can theoretically continue handling this request by ignoring the cache failure and continuing to query embedly for the URL.

Comment (jaredkerim) [inline embedly/embedly/extract.py]: Which raises a larger question which is, if a single portion of the request fails, ie failure to read from the cache or failure to query embedly, should we return a partially satisfied query for the URLs whose data we are able to query?  In which case do we signal to the client that we were only able to find the metadata for some of the URLs?

**Comment by jaredkerim** [comment]
@oyiptong @k88hudson r?

File: embedly/embedly/extract.py (hunk 1 above)
Comment (oyiptong) [inline embedly/embedly/extract.py]: I think that if there is a partial failure, we should return a partial result.
There is the situation where
In which case, the client should know that some urls weren't return given the returned data schema.

Comment (oyiptong) [inline embedly/embedly/extract.py]: This is a robustness feature that should be added later. A blanket failure is appropriate right now, as a first iteration IMO.

**Pull request merged**